### ISSUE TITLE
NVDA does not correctly highlight text when reading by words in virtual documents

### ISSUE THREAD

**Issue by k-kolev1985** [nvaccess/nvda#10660]
### Steps to reproduce:
1. From NVDA Settings -> Vision, enable "Focus Highlighting" and make sure that "Highlight browse mode cursor" is also enabled in the same settings category and apply/save the changes.
2. Open a virtual document with some text (e.g. a web page).
3. Go to a paragraph of text and start reading it word by word (with CONTROL and the left/right arrow keys).

### Actual behavior:
NVDA highlights only the first character of each word.

### Expected behavior:
NVDA should highlight the whole word.

### System configuration
- Operating system: Windows 10 Pro version 1909 (build 18363.535), 64-bit, in Bulgarian with all locale settings set to "Bulgarian".
- NVDA version: alpha-19460,d7229eff, in bulgarian.
- Processor: Intel Core i5-9400F at 2.90GHz.
- RAM Memory: 16.00GB.
- Graphics: MSI GeForce GTX-1050TI Gaming X 4G, 4096MB dedicated memory, desktop resolution set to 1920x1080.
- Sound Card: Realtek ALC892 at Intel Cannon Point PCH.

#### Does the issue still occur after restarting your PC?
Yes.

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
It has been the same ever since the "highlighting" feature was introduced.

**Comment by Adriani90** [comment]
@k-kolev1985 are you still having this issue in NVDA 2024.3 Beta?

cc: @seanbudd

**Comment by k-kolev1985** [comment]
@Adriani90 Yes, I still do have the problem, even in the latest alpha snapshots.
